workflow/util/util.go

```diff
@@ -627,13 +632,41 @@ func RetryWorkflow(kubeClient kubernetes.Interface, repo sqldb.OffloadNodeStatus
 	newWF := wf.DeepCopy()
 	podIf := kubeClient.CoreV1().Pods(wf.ObjectMeta.Namespace)
 
+	// Get all children of nodes that match filter
+	nodeIDsToReset := make(map[string]bool)
+	if restartSuccessful && len(nodeFieldSelector) > 0 {
+		selector, err := fields.ParseSelector(nodeFieldSelector)
+		if err != nil {
+			return nil, err
+		} else {
+			for _, node := range wf.Status.Nodes {
```

Comment:
Hey @mark9white, because of https://github.com/argoproj/argo/pull/2645 this actually needs to be moved further down the code. Workflows with offloaded nodes are only retrieved starting in line 678, so if a Workflow has offloaded nodes, `wf.Status.Nodes` will be nil at this point in the code and the node field selector will have no effect.

While you're at this, would you mind extracting this block out to a helper function? `RetryWorkflow` is already a bit cluttered 🙂

Reply:
Done